src/Client/Core/Entites/EntityQuery.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.DurableTask.Client.Entites;
+
+/// <summary>
+/// A query for fetching entities.
+/// </summary>
+public record EntityQuery
+{
+    /// <summary>
+    /// The default page size.
+    /// </summary>
+    public const int DefaultPageSize = 100;
+
+    /// <summary>
+    /// Gets the optional name of the entity to query for.
+    /// </summary>
+    public string? Name { get; init; }
+
+    /// <summary>
```

Comment:
I think we should add a `string?`  field `InstanceIdPrefix` because there have been situtations where we need more precise queries. (I have had users ask for this). Our backends support this already, so it will not create extra work for us, and it is not easy to change this later.

Reply:
I didn't include `InstanceIdPrefix` because `EntityQuery.Name` already translates to an orchestration query instance ID prefix. Is the prefix you suggest different? Is it the prefix of the entity key?

Or is the idea to make `Name` more robust, and give it an "equals" or "startswith" capabilities?

Reply:
The InstanceIdPrefix is more expressive because one can create prefixes of the entity name, or full entity names plus prefixes of the entity key.

For example,
1.  if the entities have instance ids where the entity name follows some convention, e.g. look something like @ModuleX_User@X  @ModuleX_Account@Z
then we can retrieve all entities whose name starts with ModuleX by specifying the prefix "@ModuleX"
2. if the entity keys have some internal structure e.g. @Task@a.b, ... @Task@a.c then we can retrieve a subset of the entities using the prefix @Task@a

Reply:
> and give it an "equals" or "startswith" capabilities?

something also worth considering I suppose. Did not think that far.